### PR TITLE
Upgrade to Spring Boot 3

### PR DESCRIPTION
**Pull request by jamesrferguson1**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/DFPL-522

### Change description ###

Upgrade to Spring Boot 3

**Does this PR introduce a breaking change?** (check one with "x")

```
[ x] Yes
[ ] No
```

### CODE REVIEW

File: charts/fpl-case-service/values.yaml
Comment (DanCatchpole) [inline charts/fpl-case-service/values.yaml:1]: will likely be some conflicts on this with the new way of doing secrets in master

File: service/build.gradle [hunk 1]
```diff
@@ -300,67 +302,60 @@ dependencies {
...
+  implementation group: 'ch.qos.logback', name: 'logback-classic', version: versions.logbackVersion
+  implementation group: 'ch.qos.logback', name: 'logback-core', version: versions.logbackVersion
+
+  implementation group: 'com.microsoft.azure', name: 'applicationinsights-core', version: '3.4.12'
+  implementation group: 'com.microsoft.azure', name: 'applicationinsights-web', version: '3.4.12'
+
+  implementation 'org.postgresql:postgresql:42.7.3'
 
-  testImplementation group: 'org.pitest', name: 'pitest', version: '1.16.1'
+  testImplementation group: 'org.pitest', name: 'pitest', version: '1.15.3'
```
Comment (DanCatchpole) [inline service/build.gradle:358]: downgrade of pitest

File: service/src/main/java/uk/gov/hmcts/reform/fpl/config/feign/codec/IdamErrorDecoder.java [hunk 2]
```diff
@@ -31,7 +33,7 @@ public Exception decode(String methodKey, Response response) {
                 response.reason(),
                 response.request().httpMethod(),
                 exception,
-                null,
+                (Date) null,
```
Comment (DanCatchpole) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/config/feign/codec/IdamErrorDecoder.java:36]: is this required? don't think I've ever seen casting of a null before!

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/ManageHearingsControllerVacatedSubmittedTest.java [hunk 3]
```diff
@@ -65,7 +65,7 @@ void shouldTriggerSendNoticeOfHearingVacatedEventWhenNotBeingRelisted() {
 
         postSubmittedEvent(caseDetails);
 
-        verify(eventPublisher, times(3)).publishEvent(sendNoticeOfHearingVacatedCaptor.capture());
+        verify(eventPublisher, times(1)).publishEvent(sendNoticeOfHearingVacatedCaptor.capture());
```
[the remark this answers is not in the document]
Comment (jamesrferguson1) [reply service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/ManageHearingsControllerVacatedSubmittedTest.java:68]: The ArgumentCaptor behaviour has changed in spring boot 3. Previously it used to capture all events regardless of the object type but now it only captures those of the specified type i.e. ArgumentCaptor<SendNoticeOfHearingVacated> will only capture events of type SendNoticeOfHearingVacated.

**Review by DanCatchpole — APPROVED** [review]

**Pull request merged**